fileAttributeCaching/src/main/kotlin/com/pkware/java/nio/file/fileAttributeCaching/ExpirableCache.kt

```diff
@@ -0,0 +1,69 @@
+package com.pkware.java.nio.file.fileAttributeCaching
+
+import java.time.Clock
+import java.util.function.Function
+
+/**
+ * TODO document.
+ */
+class ExpirableCache <A> (private val flushInterval: Long) {
+    private val clock = Clock.systemUTC()
+    private var lastFlushTime = clock.millis()
+    private val cache = HashMap<String, A?>()
+
+    /**
+     * TODO document.
+     */
+    val size: Int
+        get() = cache.size
+
+    /**
+     * TODO document.
+     */
+    fun computeIfExpiredOrAbsent(
+        key: String,
+        mappingFunction: Function<in String?, out A?>
+    ): A? {
+        recycle()
+        return cache.computeIfAbsent(key, mappingFunction)
```

Comment:
Why do a `computeIfAbsent` and not just a `putIfAbsent`? 

That way we can just say `expirableCache.putIfAbsent("key", "value")` like most map API's operate.

Reply:
We need computeIfAbsent to use the mappingFunction. The mappingFunction allows us to call out the delegate filesystemProvider if and only if the value in the cache is absent. 

This is adventageous as we want to cut down on file attribute reads/writes as much as possible to reduce OTHER_IOPS. 

Also this method allows for always getting the value from the cache and never directly from the delegate (the cache will always be repopulated)